### DISCUSSION TITLE
Scraping in Real Time from Live Market

### DISCUSSION THREAD

**Discussion by robinnarsinghranabhat** [basic-bgnr/NepseUnofficialApi#10] [Ideas]
Hi. Awesome project. 
I am using this to scrape floorsheets data from nepse daily :)

Could we add features that can help towards building a trading bot. How hard could this be .. 
I could gladly support but I have no idea on getting started. 

I am still amazed how you figured this authentication thing, ailey ko NepseAPI ma ni. I thought just copy pasting bearer token would be enough.

**Comment by basic-bgnr** [comment]
hi! @robinnarsinghranabhat. I didn't have access to my PC for few days so I couldn't reply to you earlier. 

> Could we add features that can help towards building a trading bot. How hard could this be ..
I could gladly support but I have no idea on getting started.

You would exclusively need to access [nepsetms.com.np]() API for building a trading bot, this repo won't be much helpful to you in that matter.

As per my understanding, currently [nepsetms.com.np]() backend  isn't built to support automated trading, you can check it  yourself.  for instance, TMS orderbook api doesn't return `transaction id` immediately after placing buy/sell order making automated trading difficult.

And the most important thing is
* both nepalstock and tms prohibits scraping and automated access to their website as illegal activity. 
* nepalstock access doesnt requires login, so legality of automated access cannot be enforced through user identification.
* but tms requires user login, you can be under legal trouble if something odd were to happen.

Saying all this, You can certainly reverse engineer the API but doing so would be risky (YOLO right....?) and building a trading bot around reversed API would also be very brittle.  

**Releasing public access to tms api could have severe impact on tms's load capacity and can have huge financial implication, so if you plan on doing so, please keep your repository private 👍**
